src/resources/ApiKeys/ApiKeys.ts

```diff
@@ -31,4 +31,12 @@ export default class ApiKey extends Resource {
     delete(apiKeyId: string) {
         return this.api.delete(`${ApiKey.baseUrl}/${apiKeyId}`);
     }
+
+    extend(apiKeyId: string) {
+        return this.api.put(`${ApiKey.baseUrl}/${apiKeyId}/activation/extend`);
+    }
+
+    duplicate(apiKeyId: string) {
+        return this.api.post(`${ApiKey.baseUrl}/${apiKeyId}/duplicate`);
```

Comment:
[The duplicate call](https://platformdev.cloud.coveo.com/docs/?urls.primaryName=AuthorizationServer#/Api%20Keys/rest_organizations_paramId_apikeys_paramId_duplicate_put) is a PUT, and it returns the ApiKeyModel

```suggestion
        return this.api.put<ApiKeyModel>(`${ApiKey.baseUrl}/${apiKeyId}/duplicate`);
```